### PR TITLE
Document `Tag` versioning strategy

### PR DESCRIPTION
**Pull request by axelson**
#### Changes                                                                                           
                                                                                                       
Mention the `Tag` versioning strategy                                                                  
                                                                                                       
#### Checklist                                                                                         
                                                                                                       
                                                       
                                                                                                       
- [x] Read the contribution [guide](../blob/main/CONTRIBUTING.md) and accept the [code](../blob/main/  CODE_OF_CONDUCT.md) of conduct                                                                         
- [x] Readme (updated or not needed)                                                                   
- [x] Tests (added, updated or not needed)

### CODE REVIEW

**Review by GabLeRoux — APPROVED** [review]

**Review by davidmfinol — APPROVED** [review]
Thanks, @axelson !

**Comment by webbertakken** [comment]
Note that this used to be documented, but then I removed it because too many people were confusing it with the automatic versioning option, which we are recommending for most cases.

It solved the problem, at the cost of an unclarity.

Maybe we can come up with a good way to indicate the subtle difference and reduced complexity of "auto" when compared to "tag" option.

**Comment by davidmfinol** [comment]
> Maybe we can come up with a good way to indicate the subtle difference and reduced complexity of "auto" when compared to "tag" option.
Maybe we could do the same thing that is done for the "custom" option: Label it with **(advanced users)**
We could also add a line indicating why "semantic" is recommended over "tag".

**Comment by axelson** [comment]
Ah, that's interesting. In that case I think marking it as "advanced" makes sense.

As a side-note I am getting this warning from the build (even though I'm not attempting to build for Android right now):
```
 v0.0.98

Warning: Could not parse "0.0.98
" to semver, defaulting android version code to 1
```

Although that seems like it should be a separate issue. From looking at the code I believe that there is a trailing newline that should be removed before passing the value on.

**Comment by webbertakken** [comment]
I think "Advanced" or "this requires to manage versions manually, using other actions" would be helpful. 👍

The warning indeed looks to be a bug related to a newline.

File: docs/github/v2/04-builder.md [hunk 1]
```diff
@@ -196,6 +196,15 @@ How it works:
 
 No configuration required.
 
+##### Tag
+
+Generates the version based on the latest git tag. For example `0.17.2` will
+generate the version `0.17.2`. A leading "v" is stripped, so a version of
+`v1.3.3` will generate a version of `1.3.3`.
```
Comment (davidmfinol) [inline docs/github/v2/04-builder.md:203]:
```suggestion
`v1.3.3` will generate a version of `1.3.3`. **(advanced users)**
```

**Pull request merged**